Dockerfile

```diff
@@ -19,33 +19,80 @@ ARG GOTESTSUM_VERSION=v1.9.0
 ARG GO_VERSION=1.20
 ARG ALPINE_VERSION=3.18
 
+# ALPINE_CACHE defaults to the "alpine-cache-local" stage in this image.
+# Can be overridden to a custom image for reproducible builds.
+ARG ALPINE_CACHE=alpine-cache-local
+
 # minio for s3 integration tests
 FROM minio/minio:${MINIO_VERSION} AS minio
 FROM minio/mc:${MINIO_MC_VERSION} AS minio-mc
 
 # alpine base for buildkit image
 # TODO: remove this when alpine image supports riscv64
-FROM alpine:${ALPINE_VERSION} AS alpine-amd64
-FROM alpine:${ALPINE_VERSION} AS alpine-arm
-FROM alpine:${ALPINE_VERSION} AS alpine-arm64
-FROM alpine:${ALPINE_VERSION} AS alpine-s390x
-FROM alpine:${ALPINE_VERSION} AS alpine-ppc64le
-FROM alpine:edge@sha256:2d01a16bab53a8405876cec4c27235d47455a7b72b75334c614f2fb0968b3f90 AS alpine-riscv64
+FROM --platform=amd64   alpine:${ALPINE_VERSION} AS alpine-amd64
```

Comment:
(This change is due to:
- https://github.com/moby/buildkit/issues/4237
)

Reply:
I think you need to set the base first with build platform:

```dockerfile
FROM --platform=$BUILDPLATFORM alpine:${ALPINE_VERSION} AS alpine
FROM alpine AS alpine-amd64
...
```

Pretty much like https://github.com/moby/moby/pull/44735/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R40-R52